tests/fixtures/tests/certificate-pascal-case-example.php

```diff
@@ -43,17 +45,16 @@ public function store_uses_form_request_validation()
      */
     public function store_saves()
     {
-        $certificate = $this->faker->word;
+        $certificate = factory(Certificate::class)->make();
 
-        $response = $this->post(route('certificate.store'), [
-            'certificate' => $certificate,
-        ]);
+        $response = $this->post(route('certificate.store'), $certificate->toArray());
```

Comment:
This feels a little too streamlined. It won't always be the case that all the model fields are passed for `store`.

I feel the appropriate fix is to determine the model and loop over its columns and pass them. This will give the developer more flexibility when modifying the tests.

Reply:
To be clear: you want me to replace this with the *content* of the factory, e.g. an array of ['column-name' => $faker->data, /*...*/] instead of letting the factory take care of it, correct?

Reply:
Okay, so if I do that, then we run in to the problem that the tests fail, due to the fact that the `FormRequest` is checking if the corresponding foreign id for `certificate_type_id` exists in the `certificate_types` table (which was kind of the point in #209).

The faker that is generated in the test:
```php
$certificate_type_id = $this->faker->randomDigitNotNull;
```
Which is way too naive, because there's a relationship in place. Which is why the `FormRequest` respectfully declines with a `ValidationException: The given data was invalid.`

Note: this is equally the same when you do not use the `resource: api` shorthand and if you fully type out that `validate:` statement.

Yet, The factory *knows* of this relation:
```php
$factory->define(Certificate::class, function (Faker $faker) {
    return [
        'name' => $faker->name,
        'certificate_type_id' => factory(\App\CertificateType::class),
        'reference' => $faker->word,
        'document' => $faker->word,
        'expiry_date' => $faker->date(),
        'remarks' => $faker->text,
    ];
});
```
And what if that related model has foreign id's of its own that need to exist?

Factories help setting up the world, which is especially useful in testing scenarios.

What would be your suggestion in this case?

Reply:
I think my original intention would have been to do something like this much higher up.

What we're really talking about here is a _shorthand_ for the `validate` statement that if you pass it a model, it validates "all" fields (not id + timestamps).

All of the generators work as intended. So this is really only an issue because now we're trying to implement these one off checks in each of the generators.

My solution would be for the `StatementLexer` to _expand_ a _model reference_ passed to the `validate` statement in the columns. Then let the form requests, tests, etc behave normally. I think that would get things pretty close, if not correct.